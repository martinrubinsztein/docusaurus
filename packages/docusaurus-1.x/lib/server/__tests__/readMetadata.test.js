/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

const {readSidebar} = require('../readMetadata');
const sidebarSubcategories = require('./__fixtures__/sidebar-subcategories');
const sidebarMultilevelSubcategories = require('./__fixtures__/sidebar-multilevel-subcategories');

jest.mock('../env', () => ({
  translation: {
    enabled: true,
    enabledLanguages: () => [
      {
        enabled: true,
        name: 'English',
        tag: 'en',
      },
      {
        enabled: true,
        name: '한국어',
        tag: 'ko',
      },
    ],
  },
  versioning: {
    enabled: true,
    defaultVersion: '1.0.0',
  },
}));

jest.mock(`${process.cwd()}/siteConfig.js`, () => ({}), {virtual: true});
jest.mock(`${process.cwd()}/sidebars.json`, () => true, {virtual: true});

describe('readMetadata', () => {
  describe('readSidebar', () => {
    test('should verify sub category data and verify order', () => {
      const items = readSidebar(sidebarSubcategories);
      expect(items).toMatchSnapshot();
    });
    test('should verify multilevel sub category data and verify order', () => {
      const items = readSidebar(sidebarMultilevelSubcategories);
      expect(items).toMatchSnapshot();
    });
  });
});
