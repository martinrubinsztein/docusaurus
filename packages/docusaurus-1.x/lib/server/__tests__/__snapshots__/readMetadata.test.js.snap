// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`readMetadata readSidebar should verify multilevel sub category data and verify order 1`] = `
Object {
  "doc1": Object {
    "category": "First Category",
    "next": "doc2",
    "order": 1,
    "parentSubcategory": null,
    "previous": null,
    "sidebar": "docs",
    "subcategory": null,
  },
  "doc2": Object {
    "category": "First Category",
    "next": "doc3",
    "order": 2,
    "parentSubcategory": null,
    "previous": "doc1",
    "sidebar": "docs",
    "subcategory": null,
  },
  "doc3": Object {
    "category": "Second Category",
    "next": "doc4",
    "order": 3,
    "parentSubcategory": null,
    "previous": "doc2",
    "sidebar": "docs",
    "subcategory": null,
  },
  "doc4": Object {
    "category": "Second Category",
    "next": "doc5",
    "order": 4,
    "parentSubcategory": null,
    "previous": "doc3",
    "sidebar": "docs",
    "subcategory": "First Subcategory",
  },
  "doc5": Object {
    "category": "Second Category",
    "next": "doc6",
    "order": 5,
    "parentSubcategory": null,
    "previous": "doc4",
    "sidebar": "docs",
    "subcategory": null,
  },
  "doc6": Object {
    "category": "Third Category",
    "next": "doc7",
    "order": 6,
    "parentSubcategory": null,
    "previous": "doc5",
    "sidebar": "docs",
    "subcategory": "Second Subcategory",
  },
  "doc7": Object {
    "category": "Third Category",
    "next": "doc8",
    "order": 7,
    "parentSubcategory": "Second Subcategory",
    "previous": "doc6",
    "sidebar": "docs",
    "subcategory": "Third Subcategory",
  },
  "doc8": Object {
    "category": "Third Category",
    "next": "doc9",
    "order": 8,
    "parentSubcategory": "Second Subcategory",
    "previous": "doc7",
    "sidebar": "docs",
    "subcategory": "Third Subcategory",
  },
  "doc9": Object {
    "category": "Third Category",
    "next": null,
    "order": 9,
    "parentSubcategory": null,
    "previous": "doc8",
    "sidebar": "docs",
    "subcategory": "Second Subcategory",
  },
}
`;

exports[`readMetadata readSidebar should verify sub category data and verify order 1`] = `
Object {
  "doc1": Object {
    "category": "First Category",
    "next": "doc2",
    "order": 1,
    "parentSubcategory": null,
    "previous": null,
    "sidebar": "docs",
    "subcategory": null,
  },
  "doc2": Object {
    "category": "First Category",
    "next": "doc3",
    "order": 2,
    "parentSubcategory": null,
    "previous": "doc1",
    "sidebar": "docs",
    "subcategory": null,
  },
  "doc3": Object {
    "category": "Second Category",
    "next": "doc4",
    "order": 3,
    "parentSubcategory": null,
    "previous": "doc2",
    "sidebar": "docs",
    "subcategory": null,
  },
  "doc4": Object {
    "category": "Second Category",
    "next": "doc5",
    "order": 4,
    "parentSubcategory": null,
    "previous": "doc3",
    "sidebar": "docs",
    "subcategory": "First Subcategory",
  },
  "doc5": Object {
    "category": "Second Category",
    "next": "doc6",
    "order": 5,
    "parentSubcategory": null,
    "previous": "doc4",
    "sidebar": "docs",
    "subcategory": null,
  },
  "doc6": Object {
    "category": "Third Category",
    "next": "doc7",
    "order": 6,
    "parentSubcategory": null,
    "previous": "doc5",
    "sidebar": "docs",
    "subcategory": "Second Subcategory",
  },
  "doc7": Object {
    "category": "Third Category",
    "next": "doc8",
    "order": 7,
    "parentSubcategory": null,
    "previous": "doc6",
    "sidebar": "docs",
    "subcategory": "Second Subcategory",
  },
  "doc8": Object {
    "category": "Third Category",
    "next": null,
    "order": 8,
    "parentSubcategory": null,
    "previous": "doc7",
    "sidebar": "docs",
    "subcategory": "Third Subcategory",
  },
}
`;
